documentation/create_overview_doc/createdoc.sh

```diff
@@ -570,46 +340,25 @@ function main() {
         fi
     done
 
-    # Extract content and call Bito for a system introduction and summary
-    local introduction_and_summary=$(extract_module_names_and_associated_objectives_then_call_bito "$aggregated_md_file" "$prompt_folder/system_introduction_prompt.txt")
+    local introduction_and_summary=$(extract_module_details_and_call_bito "$aggregated_md_file" "$prompt_folder/system_introduction_prompt.txt")
 
-    # Prepend the introduction and summary to the aggregated markdown file
-    # Save the current content of the aggregated file temporarily
+    # Prepend the introduction and summary to the aggregated Markdown file
     local temp_file=$(mktemp)
     mv "$aggregated_md_file" "$temp_file"
-
-    # Create a new aggregated file starting with the Markdown-formatted introduction title
     echo -e "# Introduction :\n" > "$aggregated_md_file"
-
-    # Append the introduction and summary
     echo "$introduction_and_summary" >> "$aggregated_md_file"
 
-    # Call generate_mdd_overview function here, after all .mdd files are created
-    generate_mdd_overview "$docs_folder" "$aggregated_md_file"
-    
-    # Append the content of overview.mdd to the aggregated file extracting only the Mermaid diagram block
-    if [ -f "$docs_folder/overview.mdd" ]; then
-        echo -e "\n# Full System Overview\n" >> "$aggregated_md_file"
-
-        # Extract and append only the Mermaid diagram block
-        awk '/^```mermaid$/,/^```$/' "$docs_folder/overview.mdd" >> "$aggregated_md_file"
-    else
-        echo "Overview file not found or empty."
-    fi
+    # Generate and append the system flow chart
+    generate_flow_chart_for_system "$folder_to_document" "$docs_folder/flowmaps/system_flowchart.png"
+    echo -e "\n# System Flowchart\n\n![System Flowchart](./flowmaps/system_flowchart.png)\n" >> "$aggregated_md_file"
 
     # Append the rest of the original aggregated content
     cat "$temp_file" >> "$aggregated_md_file"
-
-    # Remove the temporary file
     rm "$temp_file"
 
-    # Generate Mermaid diagrams for visual representations overwriting the markdown file with the diagrams
-    generate_mermaid_diagram "$aggregated_md_file"
-
-    # Log Session Duration and Total Token Usage to the log file
+    # Log session duration and token usage
     log_token_usage_and_session_duration
 
-    # Notify the user that the documentation has been generated
     echo "Documentation generated in $docs_folder"
```

Comment:
**Performance Issue**: Logging session duration and token usage at the end of the script does not account for early exits or errors that may occur, potentially leading to incomplete or inaccurate logging. <br> **Fix**: Implement a more comprehensive logging strategy that accounts for early exits due to errors. This could include setting a trap on EXIT or ERROR signals to ensure logging occurs regardless of how the script terminates. <br> **Code Suggestion**: 
 ```
 +    trap 'log_token_usage_and_session_duration' EXIT
 +    # Existing script content...
 ```